models/aavegotchi/polygon/aavegotchi_polygon_events.sql

```diff
@@ -0,0 +1,190 @@
+{{ config(
+    schema = 'aavegotchi_polygon',
+    alias = 'events',
+    partition_by = ['block_date'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_time', 'unique_trade_id'],
+    post_hook='{{ expose_spells(\'["polygon"]\'
+                              "project",
+                              "aavegotchi",
+                              \'["springzh"]\') }}'
+    )
+}}
+
+{% set nft_start_date = "2021-03-02" %}
+
+WITH contract_list as (
+    SELECT distinct erc721TokenAddress as nft_contract_address
+    FROM aavegotchi_polygon.aavegotchi_diamond_evt_ERC721ExecutedListing
+    UNION ALL
+    SELECT distinct erc1155TokenAddress as nft_contract_address
+    FROM aavegotchi_polygon.aavegotchi_diamond_evt_ERC1155ExecutedListing
+),
+
+mints as (
+    SELECT 'mint' AS trade_category,
+        block_time AS evt_block_time,
+        block_number AS evt_block_number,
+        tx_hash AS evt_tx_hash,
+        NULL AS contract_address, -- We eave it NULL here and will get its value below by join from transactions table
+        evt_index,
+        'Mint' AS evt_type,
+        `to` AS buyer,
+        NULL AS seller,
+        contract_address AS nft_contract_address,
+        token_id,
+        amount AS number_of_items,
+        token_standard,
+        '0x385eeac5cb85a38a9a07a70c73e0a3271cfb54a7' AS currency_contract, -- All sale are in GHST
+        CAST(0 as DECIMAL(38,0)) AS amount_raw, -- It's hard to get the mint price. So handle it similar as in nftb_bnb_events
+        NULL AS category,
+        NULL AS executed_time
```

Comment:
We can cast null to their expected types

Reply:
Updated.